book/src/input/handling_input.md

```diff
@@ -0,0 +1,144 @@
+# Handling Input
+
+Amethyst uses an `InputHandler` to handle user input.
+You initialise this `InputHandler` by creating an `InputBundle` and adding it to the game data.
+
+```rust,edition2018,no_run,noplaypen
+use amethyst{
+  prelude::*,
+  utils::application_root_dir,
+  input::{InputBundle, StringBindings},
+};
+
+fn main() -> amethyst::Result<()> {
+  // StringBindings is the default BindingTypes
+  let input_bundle = InputBundle::<StringBindings>::new();
+
+  let game_data = GameDataBuilder::default()
+    //..
+    .with_bundle(input_bundle)?
+    //..
+
+  //..
```

Comment:
Yeah, I wasn't sure how to go about that. Do I have to write the full code in one of the examples and remove the 'norun'?

Reply:
perhaps not all, but the ones that would be annoying to readers when it goes out of date.
A soft rule is to have at least one tested snippet on each page, so that when it gets out of date with the code, CI would fail, and we should know to check that page's snippets.

side story: I tend to work with a full example, and just hide / remove bits that aren't needed in each snippet.

Reply:
Well I do have the full code somewhere, so I will try to make that work. Do I just use `mdbook test` to test if CI builds?

Reply:
oh, you have to go:

```bash
# in amethyst repo
mdbook test -L target/debug/deps book
```

if it complains that it can't find a crate to link to (usually "multiple matching crates" or something), try:

```bash
rm -rf ./target/debug/deps/libamethyst*
cargo test --all --no-run --features empty
# then
mdbook test -L target/debug/deps book
```

Reply:
Oh ya, it takes a while to test the whole book locally, I tend to remove all the other pages from the `SUMMARY.md` and just test the ones I've changed.

Reply:
I made the examples build, but it only attempted to compile them when I removed the `edition2018`. It also seemed the same for all the other examples in the book.